src/framework/mlt_property.c

```diff
@@ -1279,7 +1281,50 @@ int mlt_property_interpolate( mlt_property self, mlt_property p[],
 	double progress, double fps, locale_t locale, mlt_keyframe_type interp )
 {
 	int error = 0;
-	if ( interp != mlt_keyframe_discrete &&
+	if ( interp != mlt_keyframe_discrete && ( self->types & mlt_prop_color ) )
+	{
+		mlt_color value = { 0xff, 0xff, 0xff, 0xff };
+		if ( interp == mlt_keyframe_linear )
+		{
+			mlt_color colors[2];
+			mlt_color zero = { 0xff, 0xff, 0xff, 0xff };
+			colors[0] = p[1]? mlt_property_get_color( p[1], fps, locale ) : zero;
+			if ( p[2] )
+			{
+				colors[1] = mlt_property_get_color( p[2], fps, locale );
+				value.r = linear_interpolate( colors[0].r, colors[1].r, progress );
+				value.g = linear_interpolate( colors[0].g, colors[1].g, progress );
+				value.b = linear_interpolate( colors[0].b, colors[1].b, progress );
+				value.a = linear_interpolate( colors[0].a, colors[1].a, progress );
+			}
+			else
+			{
+				value = colors[0];
+			}
+		}
+		else if ( interp == mlt_keyframe_smooth )
+		{
+			mlt_color colors[4];
+			mlt_color zero = { 0xff, 0xff, 0xff, 0xff };
+			colors[1] = p[1]? mlt_property_get_color( p[1], fps, locale ) : zero;
+			if ( p[2] )
+			{
+				colors[0] = p[0]? mlt_property_get_color( p[0], fps, locale ) : zero;
+				colors[2] = p[2]? mlt_property_get_color( p[2], fps, locale ) : zero;
+				colors[3] = p[3]? mlt_property_get_color( p[3], fps, locale ) : zero;
+				value.r = catmull_rom_interpolate( colors[0].r, colors[1].r, colors[2].r, colors[3].r, progress );
+				value.g = catmull_rom_interpolate( colors[0].g, colors[1].g, colors[2].g, colors[3].g, progress );
+				value.b = catmull_rom_interpolate( colors[0].b, colors[1].b, colors[2].b, colors[3].b, progress );
+				value.a = catmull_rom_interpolate( colors[0].a, colors[1].a, colors[2].a, colors[3].a, progress );
```

Comment:
You need to clamp these values to [0,255]. Drawing a curve through a set of points does not understand value ranges.

Reply:
Done, and the issues I noticed seem to be fixed. Thanks for the hint.